File: day2/Python/part1.py
import re
with open("input.txt", "r") as file:
    contents = file.read()
line_array = re.split("\n",contents) 
horizontal = 0
depth = 0
for line in line_array:
    array = re.split(" ",line)
    direction = array[0]
    num = int(array[1])
    if direction == "forward":
        horizontal += num
    elif direction == "down":
        depth += num
    elif direction == "up":
        depth -= num
print(f"Total is {horizontal * depth}")    

File: day2/JS/part1.js
const lineReader = require("line-reader");

let horizontal = 0;
let depth = 0;

lineReader.eachLine("input.txt", (line,last) => {
  parseLine(line);
  if (last == true)
    console.log(horizontal * depth);
});

function parseLine(text) {
  let arr = text.split(" ");
  arr[1] = parseInt(arr[1]);
  switch (arr[0]) {
    case "forward":
      horizontal += arr[1];
      break;
    case "down":
      depth += arr[1];
      break;
    case "up":
      depth -= arr[1];
      break;
  }
}

